.github/workflows/test-build-deploy.yml

```
@@ -113,58 +113,30 @@ jobs:
         uses: actions/checkout@v2
       - name: Install Docker Client
         run: ./.github/workflows/scripts/install-docker.sh
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
       - name: Symlink Expected Path to Workspace
         run: |
           mkdir -p /go/src/github.com/grafana/mimir
           ln -s $GITHUB_WORKSPACE/* /go/src/github.com/grafana/mimir
-      - name: Build Image
+      - name: Build Multiarch Docker Images Locally
+        # Ignore mimir-build-image and mimir-rules-action.
         run: |
-          make BUILD_IN_CONTAINER=false ./cmd/mimir/.uptodate
-      - name: Save Images
+          ./.github/workflows/scripts/build-images.sh /tmp/images $(make list-image-targets | grep -v -E '/mimir-build-image/' | grep -v -E '/mimir-rules-action/')
```

[nit] I think we can simplify it.

```suggestion
          ./.github/workflows/scripts/build-images.sh /tmp/images $(make list-image-targets | grep -v -E '/mimir-build-image|mimir-rules-action/')
```